patroni/ctl.py

```diff
@@ -436,25 +484,58 @@ def ctl_load_config(cluster_name, config_file, dcs):
 @click.option('--role', '-r', help='Restart only members with this role', default='any',
               type=click.Choice(['master', 'replica', 'any']))
 @click.option('--any', 'p_any', help='Restart a single member only', is_flag=True)
+@click.option('--scheduled', help='Timestamp of a scheduled restart in unambiguous format (e.g. ISO 8601)',
+              default=None)
+@click.option('--pg-version', 'version', help='Restart if the PostgreSQL version is less than provided (e.g. 9.5.2)',
+              default=None)
+@click.option('--pending', help='Restart if pending', is_flag=True)
 @option_config_file
 @option_force
 @option_dcs
-def restart(cluster_name, member_names, config_file, dcs, force, role, p_any):
+def restart(cluster_name, member_names, config_file, dcs, force, role, p_any, scheduled, version, pending):
     config, dcs, cluster = ctl_load_config(cluster_name, config_file, dcs)
 
-    role_names = [m.name for m in get_all_members(cluster, role)]
-
-    if member_names:
-        member_names = list(set(member_names) & set(role_names))
-    else:
-        member_names = role_names
+    members = get_members(cluster, cluster_name, member_names, role, force, 'restart')
 
     if p_any:
-        random.shuffle(member_names)
-        member_names = member_names[:1]
+        k = random.choice(list(members))
+        members = {k: members[k]}
 
-    output_members(cluster, cluster_name)
-    empty_post_to_members(cluster, member_names, force, 'restart', auth_header(config))
+    if version is None and not force:
+        version = click.prompt('Restart if the PostgreSQL version is less than provided (e.g. 9.5.2) ',
+                               type=str, default='')
+
+    content = {}
+    if pending:
+        content['restart_pending'] = True
+
+    if version:
+        if not re.match(r'[1-9][0-9]?(\.(0|([1-9][0-9]?))){2}$', version):
```

Comment:
I wonder, should we create function which is doing version check, put it into utils and use it in the api.py and ctl.py to avoid code duplication and improve maintainability?